secure_tunneling/source/SecureTunnel.cpp

```diff
@@ -125,12 +168,35 @@ namespace Aws
 
         const Crt::Optional<Crt::ByteCursor> &Message::getServiceId() const noexcept { return m_serviceId; }
 
+        const uint32_t &Message::getConnectionId() const noexcept { return m_connectionId; }
+
         Message::~Message()
         {
             aws_byte_buf_clean_up(&m_payloadStorage);
             aws_byte_buf_clean_up(&m_serviceIdStorage);
         }
 
+        //***********************************************************************************************************************
+        /*                                              SendMessageCompleteData */
+        //***********************************************************************************************************************
+
+        SendMessageCompleteData::SendMessageCompleteData(
+            enum aws_secure_tunnel_message_type type,
+            Crt::Allocator *allocator) noexcept
+            : m_allocator(allocator)
+        {
+            aws_byte_buf_clean_up(&m_messageTypeStorage);
+            AWS_ZERO_STRUCT(m_messageTypeStorage);
+            struct aws_byte_buf typeBuf = aws_byte_buf_from_c_str(aws_secure_tunnel_message_type_to_c_string(type));
```

Comment:
Does `typeBuf` need to be cleaned after the `aws_byte_buf_init_copy` call? From what I can tell, it looks like it [makes a new copy](https://github.com/awslabs/aws-c-common/blob/194b11eddf2e55854a0994e35115f4ebd988f0b3/source/byte_buf.c#L48) of the data, so I think afterwards you need to erase `typeBuf`. I'm not positive though...

Reply:
I think you're right!